### PR TITLE
Change the VID and PID of the file kb38 info.json

### PR DESCRIPTION
**Pull request by DOIO2022**
## Description

Change the VID and PID of the file kb38 info.json

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by waffle87 — APPROVED** [review]
Keyboard updates must target the `develop` branch.

**Review by fauxpark — APPROVED** [review]

**Comment by DOIO2022** [comment]
> Keyboard updates must target the `develop` branch.

@waffle87  Hi,can I change the PID and VID of the file info.json?  I want via supported, but the PID and VID is wrong format. Thank you!

File: keyboards/doio/kb38/info.json [hunk 1]
```diff
@@ -22,8 +22,8 @@
     "processor": "atmega32u4",
     "usb": {
         "device_version": "1.0.0",
-        "pid": "0x0000",
-        "vid": "0xFEED"
+        "pid": "0x3801",
+        "vid": "0xD010"
```
Comment (weswigham) [inline keyboards/doio/kb38/info.json:26]: I'm confused by this change, I first remarked on this over on the [via repo](https://github.com/the-via/keyboards/pull/2224#discussion_r1578679264), but it seems like the kb38 I have reports a `vid` of `0xFEED` (matching the old value), and a pid of `0x6060` (which doesn't match either the old or new value for this, but _does_ match the `pid` in the `via` dev json [linked on the keebmonkey wiki](https://wiki.keebmonkey.com/en/products/mmp). Is this something there's a firmware update in the works to change or something?

**Comment by DOIO2022** [comment]
Our client wants the keyboard to support VIA's automatic recognition, just like this website https://usevia.app/&nbsp;
But now the QMK environment no longer supports the current VID and PID
So&nbsp; we need change the VID and PID&nbsp; to support the new environment.






                       
原始邮件
                       
                     

发件人："Wesley Wigham"< ***@***.*** &gt;;

发件时间：2024/4/25 9:43

收件人："qmk/qmk_firmware"< ***@***.*** &gt;;

抄送人："DOIO2022"< ***@***.*** &gt;;"Author"< ***@***.*** &gt;;

主题：Re: [qmk/qmk_firmware] Change the VID and PID of the file kb38info.json (PR #23393)




 
@weswigham commented on this pull request.
 
 
In keyboards/doio/kb38/info.json:
 &gt; @@ -22,8 +22,8 @@      "processor": "atmega32u4",      "usb": {          "device_version": "1.0.0", -        "pid": "0x0000", -        "vid": "0xFEED" +        "pid": "0x3801", +        "vid": "0xD010"  
I'm confused by this change, I first remarked on this over on the via repo, but it seems like the kb38 I have reports a vid of 0xFEED (matching the old value), and a pid of 0x6060 (which doesn't match either the old or new value for this, but does match the pid in the via dev json linked on the keebmonkey wiki. Is this something there's a firmware update in the works to change or something?

**Comment by weswigham** [comment]
Right, but if these values don't match what the device actually reports for pid/vid, it won't be detected anyway, no? (At least via won't, anyway)

**Comment by weswigham** [comment]
Furthermore, the vendor id it was changed to here, 0xD010 (53264 in decimal) doesn't seem to be in the registry of valid [USB vendor IDs](https://usb.org/sites/default/files/vendor_ids051920_0.pdf), but I don't really care about that so much as it just not being what the device reports, so just not being useful. 🤷‍♂️

**Comment by DOIO2022** [comment]
Yes. So we want change the&nbsp; VID and PID of the info.json in the qmk, and then&nbsp;upload the layout to via GitHub to achieve automatic recognition



                       
原始邮件
                       
                     

发件人："Wesley Wigham"< ***@***.*** &gt;;

发件时间：2024/4/25 15:02

收件人："qmk/qmk_firmware"< ***@***.*** &gt;;

抄送人："DOIO2022"< ***@***.*** &gt;;"Author"< ***@***.*** &gt;;

主题：Re: [qmk/qmk_firmware] Change the VID and PID of the file kb38info.json (PR #23393)




 
Right, but if these values don't match what the device actually reports for pid/vid, it won't be detected anyway, no? (At least via won't, anyway)

**Pull request merged**